File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: list[int]) -> int:
        i ,res= 0 ,0
        j = len(height)-1
        while i < j:
            res = max(res,(j-i)*min(height[i],height[j]))
            if height[i] < height[j]:
                i +=1
            else:
                j -=1
        return res

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length - 1, area = 0;

        while(left < right){
            int loc_area = (right - left)*Math.min(height[left], height[right]);
            area = Math.max(area, loc_area);
            if(height[right] > height[left]){
                left++;
            }
            else{
                right--;
            }
        }
        return area;
    }
}